### DISCUSSION TITLE
Paper Pan Event

### DISCUSSION THREAD

**Discussion by CodeSippr** [clientIO/joint#2312] [Q&A] [answered]
### Introduction

Is it possible to detect whether the pan event is triggered by a mouse or a trackpad?

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS

**Comment by kumilingus** [comment]
Why? What is the use case?

**Reply by CodeSippr** [reply]
Currently I'm using this event to manage the zoom for the paper. This should remain the case with the mouse, but I would like to use the trackpad to navigate through the graph

**Reply by kumilingus** [reply, marked as answer]
This worked for me (taken from [SO](https://stackoverflow.com/a/62415754/2921495)).
```js
function detectTrackPad(e) {
    var isTrackpad = false;
    if (e.wheelDeltaY) {
        if (e.wheelDeltaY === (e.deltaY * -3)) {
            isTrackpad = true;
        }
    }
    else if (e.deltaMode === 0) {
        isTrackpad = true;
    }
    console.log(isTrackpad ? 'Trackpad detected' : 'Mousewheel detected');
}

paper.on('paper:pan', (evt, tx, ty) => {
    detectTrackPad(evt.originalEvent);
});
```
